File: XMLCheck/Handler/CheckRulesXML02/Xml2RuleRunner.cs
using XMLCheck.Handler.CheckRulesXML2;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML02
{
    public static class Xml2RuleRunner
    {
        private static readonly List<IXml2Rule> _rules = new()
        {
            // rule khác
        };

        public static List<ErrorDetails> Run(
            List<Xml2Model> models,
            HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            foreach (var m in models)
            {
                foreach (var rule in _rules)
                {
                    var err = rule.Check(m, context);
                    if (err != null)
                        errors.Add(err);
                }
            }

            return errors;
        }

    }

}

File: XMLValidator_v2/src/parsers/rules/xml2/xml2RuleRunner.ts
import { Xml2Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml2Rule } from './xml2RuleBase';

export class Xml2RuleRunner {
  private static readonly rules: IXml2Rule[] = [];

  public static run(models: Xml2Model[], context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!models || models.length === 0) return errors;

    for (const model of models) {
      for (const rule of this.rules) {
        if (!rule.isEnabled) continue;
        const err = rule.check(model, context);
        if (err) {
          errors.push(err);
        }
      }
    }

    return errors;
  }
}
